<?xml version="1.0" encoding="UTF-8"?>

<launch>
	<!-- IMU -->
	<node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
		<param name="port" value="/dev/ttyIMU" />
		<param name="baud_rate" value="115200" />
	</node>

	<node pkg="obstacle_detector" type="obstacle_detector" name="obstacle_detector" output="screen">
		<param name="world_frame"     type="string" value="velodyne"/>
		<param name="scanner_frame"   type="string" value="scanner"/>
	
		<param name="use_scan"            type="bool" value="true"/>
		<param name="use_pcl"             type="bool" value="false"/>
		<param name="transform_to_world"  type="bool" value="false"/>
		<param name="use_split_and_merge" type="bool" value="false"/>

		<param name="min_group_points"    value="3"/>
		<param name="max_group_distance"  value="0.100"/>
		<param name="distance_proportion" value="0.006136"/>
		<param name="max_split_distance"  value="0.100"/>

		<param name="max_merge_separation" value="0.200"/>
		<param name="max_merge_spread"     value="0.070"/>
		<param name="max_circle_radius"    value="0.300"/>
		<param name="radius_enlargement"   value="0.050"/>

		<param name="max_scanner_range"    value="5.0"/>
		<param name="max_x_range"          value="2.0"/>
		<param name="min_x_range"          value="-2.0"/>
		<param name="max_y_range"          value="1.0"/>
		<param name="min_y_range"          value="-1.0"/>
  	</node>

    <node pkg="obstacle_detector" type="obstacle_visualizer" name="obstacle_visualizer" output="screen"/>

	<!-- Nodes -->
	<node pkg="tricat_241" type="gnss_converter.py" name="gnss_converter" respawn="true" output="screen"/>
	<node pkg="tricat_241" type="psi_converter.py" name="psi_converter" respawn="true"/>
	<node pkg="tricat_241" type="sensor_total.py" name="sensor_total" args="Sensor_total" respawn="true"/>

	<!-- Parameters -->
	<rosparam command="load" file="$(find tricat_241)/params/241_params.yaml"/>
	<rosparam command="load" file="$(find tricat_241)/params/lidar_params.yaml"/>
</launch>

